File: FirstComeFirstServe.java
import java.util.Scanner;

class NumberErrorException extends Exception {
	/*
	 * Exception to throw when the input is not possible
	 */
	private static final long serialVersionUID = -2856377272740965184L;

	public String toString() {
		return ("Invalid number.");
	}
}

class FirstComeFirstServe {
	public static void main(String[] args) throws NumberErrorException {
		System.out.println("THE FCFS ALGO");
		Scanner mys = new Scanner(System.in);
		System.out.print("Enter total number of processes: ");
		int numProcess = mys.nextInt();
		if (numProcess <= 0)
			throw new NumberErrorException();
		int arrivalTime[] = new int[numProcess];
		int burstTime[] = new int[numProcess];
		for (int i = 0; i < numProcess; i++) {
			System.out.print("Enter Arrival time of P" + (i + 1) + ": ");
			arrivalTime[i] = mys.nextInt();
			if (arrivalTime[i] < 0)
				throw new NumberErrorException();
			System.out.print("Enter Burst time of P" + (i + 1) + ": ");
			burstTime[i] = mys.nextInt();
			if (burstTime[i] < 0)
				throw new NumberErrorException();
		}

		int completionTime[] = new int[numProcess];
		for (int i = 0; i < numProcess; i++)
			completionTime[i] = 0;
		for (int i = 0; i < numProcess; i++) {
			for (int j = 0; j <= i; j++) {
				completionTime[i] += burstTime[j];
			}
		}

		double avgTAT = 0;
		double avgWT = 0;
		int waitingTime[] = new int[numProcess];
		int turnAround[] = new int[numProcess];
		for (int i = 0; i < numProcess; i++) {
			turnAround[i] = completionTime[i] - arrivalTime[i];
			waitingTime[i] = turnAround[i] - burstTime[i];
			avgTAT += turnAround[i];
			avgWT += waitingTime[i];
		}
		avgTAT /= numProcess;
		avgWT /= numProcess;

		System.out.println("Process \tAT \t BT \t CT \t TAT \t WT");
		for (int i = 0; i < numProcess; i++) {
			System.out.println(" P" + (1 + i) + "\t\t" + arrivalTime[i] + "ms" + "\t" + burstTime[i] + "ms" + "\t"
					+ completionTime[i] + "ms" + "\t" + turnAround[i] + "ms" + "\t" + waitingTime[i] + "ms");
		}
		System.out.println("Average Turn Around Time: " + avgTAT);
		System.out.println("Average Waiting Time: " + avgWT);

	}
}

File: FirstComeFirstServe.py
print('First Come First Serve')
num_processes = int(input('Enter total number of processes: '))

arrival_time = []
burst_time = []
for i in range(0, num_processes):
  arrival_time.append(int(input('Enter arrival time of P'+ str(i+1) + ': ')))
  burst_time.append(int(input('Enter burst time of P'+ str(i+1) + ': ')))
  
completion_time = []

for i in range(0, num_processes, 1):
  completion_time.append(0)

for i in range(0, num_processes, 1):
  for j in range(0, i+1, 1):
    completion_time[i] += burst_time[j]

waiting_time = []
turnaround = []
avg_tat = 0
avg_wt = 0

for i in range(0, num_processes, 1):
  turnaround.append(completion_time[i] - arrival_time[i])
  waiting_time.append(turnaround[i] - burst_time[i])
  avg_tat += turnaround[i]
  avg_wt += waiting_time[i]

avg_tat /= num_processes
avg_wt /= num_processes

print('Process \tAT \tBT \tCT \tTAT \tWT')
for i in range(0, num_processes,1):
  print(" P" + str((1 + i)) + "\t\t" + str(arrival_time[i]) + "ms" + "\t" + str(burst_time[i]) + "ms" + "\t" +str (completion_time[i]) + "ms" + "\t" + str(turnaround[i]) + "ms" + "\t" + str(waiting_time[i]) + "ms")

print("Average Turn Around Time: " + str(avg_tat))
print("Average Waiting Time: " + str(avg_wt))
